core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.text.TextUtils
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.card.MaterialCardView
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.getAttribute
+import org.kiwix.kiwixmobile.core.extensions.tint
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getToolbarHeight
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getWindowHeight
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getWindowWidth
+import org.kiwix.kiwixmobile.core.utils.ImageUtils.getBitmapFromView
+import org.kiwix.kiwixmobile.core.utils.StyleUtils.fromHtml
+
+class TabsAdapter internal constructor(
+  private val activity: AppCompatActivity,
+  private val webViews: List<KiwixWebView>,
+  private val painter: NightModeViewPainter
+) : RecyclerView.Adapter<TabsAdapter.ViewHolder>() {
+  private var listener: TabClickListener? = null
+  var selected = 0
+  @SuppressLint("ResourceType") override fun onCreateViewHolder(
```

Comment:
its kind of a lot to ask but do you feel up to converting this to xml? So much effort for just a Textview and 2 ImageView in a CardView.

Ah I guess

```
ConstraintLayout.LayoutParams(
        activity.getWindowWidth() / 2,
        -activity.getToolbarHeight() / 2 + activity.getWindowHeight() / 2
      )
```
this is the kicker. I imagine we could affect something very similar with a custom LayoutManager but I'll admit to never having written one. Eh ignore this comment and we can just think about the future of this class later on

Reply:
When I was writing this a couple of years back I was feeling so cool. But now it feels so bad. XD

You can share if you have something in mind to get rid of this.

Reply:
If you want to create a ticket we can discuss there but for now this works so I don't feel a pressing urge to poke it